model-optimizer/unit_tests/extensions/back/ShapeOfConstFolding_test.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import unittest
+
+import numpy as np
+
+from extensions.back.ShapeOfConstFolding import ShapeOfConstFolding
+from mo.front.common.partial_infer.eltwise import eltwise_infer
+from mo.front.common.partial_infer.utils import int64_array
+from mo.utils.ir_engine.compare_graphs import compare_graphs
+from unit_tests.utils.graph import build_graph
+
+const_value = np.random.rand(1, 3, 30, 30)
+nodes_attributes = {'input': {'shape': int64_array([1, 3, 30, 30]), 'type': 'Parameter', 'kind': 'op',
+                              'op': 'Parameter'},
+                    'input_data': {'value': None, 'shape': int64_array([1, 3, 30, 30]), 'kind': 'data'},
+                    'const': {'type': 'Const', 'kind': 'op', 'op': 'Const', 'value': const_value},
+                    'const_data': {'kind': 'data', 'value': const_value},
+                    'shapeof_input': {'kind': 'op', 'op': 'ShapeOf', 'value': int64_array([1, 3, 30, 30])},
+                    'shapeof_input_data': {'value': None, 'shape': None, 'kind': 'data',
+                                           'value': int64_array([1, 3, 30, 30])},
+
+                    'shapeof_const': {'kind': 'op', 'op': 'ShapeOf', 'value': int64_array([1, 3, 30, 30])},
+                    'shapeof_const_data': {'value': None, 'shape': None, 'kind': 'data',
+                                           'value': int64_array([1, 3, 30, 30])},
+
+                    'mul': {'kind': 'op', 'op': 'Mul', 'infer': lambda node: eltwise_infer(node, lambda a, b: a * b)},
+                    'mul_data': {'kind': 'data', 'value': np.array([1, 9, 900, 900])},
+                    'last': {'kind': 'op', 'op': 'Result'},
+
+                    # new nodes
+                    'new_const_shapeof': {'type': 'Const', 'kind': 'op', 'op': 'Const',
+                                          'value': int64_array([1, 3, 30, 30])},
+                    'new_const_shapeof_data': {'value': int64_array([1, 3, 30, 30]), 'kind': 'data'},
+                    }
+
+
+class ShapeOfConstFoldingTests(unittest.TestCase):
+    def test1(self):
```

Comment:
I don't see a test case the for following graph:

![image](https://user-images.githubusercontent.com/30651952/121127488-85953400-c832-11eb-8d0c-e2cf4d60eb62.png)